servers/republik/modules/crowdfundings/lib/scheduler/owners.js

```diff
@@ -97,10 +106,24 @@ const getBuckets = async ({ now }, { pgdb }) => {
       )
         .then(date => date && moment(date))
 
-      stats.numNeedProlongProgress += 1
+      stats.numNeedProlongProgress++
 
       if (prolongBeforeDate) {
         const dropped = buckets.some(bucket => {
+          // Don't add user to bucket if user.membershipTypes does not contain
+          // any of memberships listed in bucket.onlyMembershipTypes.
+          if (bucket.onlyMembershipTypes) {
+            const hasNecessaryMembershipType = user.membershipTypes
+              .map(type => bucket.onlyMembershipTypes.includes(type))
+              .reduce((acc, cur) => !cur ? acc : cur, false)
```

Comment:
just use `!!user.membershipTypes.find...` for more clarity and to stop searching once you found.